src/components/common/ConnectWallet/MPCLogin.tsx

```diff
@@ -0,0 +1,81 @@
+import { MPCWalletState } from '@/hooks/wallets/mpc/useMPCWallet'
+import { Box, Button, SvgIcon, Typography } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
+import { MpcWalletContext } from './MPCWalletProvider'
+import { PasswordRecovery } from './PasswordRecovery'
+import GoogleLogo from '@/public/images/welcome/logo-google.svg'
+
+import css from './styles.module.css'
+import useWallet from '@/hooks/wallets/useWallet'
+
+const MPCLogin = ({ onLogin }: { onLogin?: () => void }) => {
+  const { loginPending, triggerLogin, userInfo, walletState, recoverFactorWithPassword } = useContext(MpcWalletContext)
+
+  const wallet = useWallet()
+
+  const [loginTriggered, setLoginTriggered] = useState(false)
+
+  const login = async () => {
+    setLoginTriggered(true)
+    await triggerLogin()
+  }
+
+  // If login was triggered through the Button we immediately continue if logged in
+  useEffect(() => {
+    if (loginTriggered && wallet && onLogin) {
+      onLogin()
+    }
+  }, [loginTriggered, onLogin, wallet])
```

Comment:
I don't think so.
When `triggerLogin` resolves, it might result in a `MANUAL_RECOVERY` state, which will pop up the Password recovery modal.
So we have to react to a side effect here.

Reply:
Same here: Calling the callback after triggerLogin / recoverPassword resolves causes the useSyncSafeCreationStep to immediately redirect us back.